ppx/ppx_tyxml.ml

```diff
@@ -21,6 +21,57 @@ open Asttypes
 open Parsetree
 
 
+module H = Hashtbl.Make(struct
+    type t = string
+    let hash = Hashtbl.hash
+    let equal (x:string) y = x = y
+  end)
+
+(* We replace expressions by dummy token "tyxmlX".
+   We store a table token to expression to retrieve them after parsing.
+*)
+let gen_token, find_expr =
+  let r = ref 0 in
+  let tbl = H.create 17 in
+  let make_id () = incr r ; "tyxml" ^ string_of_int !r in
```

Comment:
It seems to me that a more "distinctive" (ugly) prefix than `tyxml` would be a good idea.. just in case of conflict with something the user chooses somehow.